File: Java/Program/Komputer.java
import java.util.ArrayList;
import java.util.List;

public class Komputer {
    private String nama;
    private CPU cpu;
    private List<RAM> ramList;
    private Motherboard motherboard;
    private List<SSD> ssdList;
    private List<HDD> hddList;
    private Monitor monitor;
    private SmartSpeaker smartSpeaker;

    // Constructor
    public Komputer() {
        this.nama = "";
        this.cpu = new CPU();
        this.ramList = new ArrayList<>();
        this.motherboard = new Motherboard();
        this.ssdList = new ArrayList<>();
        this.hddList = new ArrayList<>();
        this.monitor = new Monitor();
        this.smartSpeaker = new SmartSpeaker();
    }

    public Komputer(String nama, CPU cpu, List<RAM> ramList, Motherboard motherboard,
                   List<SSD> ssdList, List<HDD> hddList, Monitor monitor, SmartSpeaker smartSpeaker) {
        this.nama = nama;
        this.cpu = cpu;
        this.ramList = ramList;
        this.motherboard = motherboard;
        this.ssdList = ssdList;
        this.hddList = hddList;
        this.monitor = monitor;
        this.smartSpeaker = smartSpeaker;
    }

    // Setter methods
    public void setNama(String nama) { this.nama = nama; }
    public void setCpu(CPU cpu) { this.cpu = cpu; }
    public void setRamList(List<RAM> ramList) { this.ramList = ramList; }
    public void setMotherboard(Motherboard motherboard) { this.motherboard = motherboard; }
    public void setSsdList(List<SSD> ssdList) { this.ssdList = ssdList; }
    public void setHddList(List<HDD> hddList) { this.hddList = hddList; }
    public void setMonitor(Monitor monitor) { this.monitor = monitor; }
    public void setSmartSpeaker(SmartSpeaker smartSpeaker) { this.smartSpeaker = smartSpeaker; }

    // Getter methods
    public String getNama() { return nama; }
    public CPU getCpu() { return cpu; }
    public List<RAM> getRamList() { return ramList; }
    public Motherboard getMotherboard() { return motherboard; }
    public List<SSD> getSsdList() { return ssdList; }
    public List<HDD> getHddList() { return hddList; }
    public Monitor getMonitor() { return monitor; }
    public SmartSpeaker getSmartSpeaker() { return smartSpeaker; }

    // Method untuk manipulasi komponen
    public void tambahRAM(RAM ram) {
        ramList.add(ram);
        System.out.println("RAM " + ram.getNama() + " berhasil ditambahkan ke " + nama);
    }

    public void tambahSSD(SSD ssd) {
        ssdList.add(ssd);
        System.out.println("SSD " + ssd.getNama() + " berhasil ditambahkan ke " + nama);
    }

    public void tambahHDD(HDD hdd) {
        hddList.add(hdd);
        System.out.println("HDD " + hdd.getNama() + " berhasil ditambahkan ke " + nama);
    }

    // Tampilkan Info
    public void tampilkanSpesifikasi() {
        System.out.println("========== SPESIFIKASI KOMPUTER: " + nama + " ==========\n");
        
        System.out.println("CPU: " + cpu.tampilkanInfo() + "\n");
        
        System.out.println("RAM:");
        for (int i = 0; i < ramList.size(); i++) {
            System.out.println("  [" + (i+1) + "] " + ramList.get(i).tampilkanInfo());
        }
        System.out.println();
        
        System.out.println("MOTHERBOARD: " + motherboard.tampilkanInfo() + "\n");
        
        System.out.println("STORAGE:");
        System.out.println("  SSD:");
        for (int i = 0; i < ssdList.size(); i++) {
            System.out.println("    [" + (i+1) + "] " + ssdList.get(i).tampilkanInfo());
        }
        
        System.out.println("  HDD:");
        for (int i = 0; i < hddList.size(); i++) {
            System.out.println("    [" + (i+1) + "] " + hddList.get(i).tampilkanInfo());
        }
        System.out.println();
        
        System.out.println("MONITOR: " + monitor.tampilkanInfo() + "\n");
        
        System.out.println("SMART SPEAKER: " + smartSpeaker.tampilkanInfo() + "\n");
        
        System.out.println("=======================================================");
    }
}

File: Python/Program/Komputer.py
from CPU import CPU
from Motherboard import Motherboard
from Monitor import Monitor
from SmartSpeaker import SmartSpeaker

class Komputer:
    def __init__(self, nama="", cpu=None, ram_list=None, motherboard=None, 
                 ssd_list=None, hdd_list=None, monitor=None, smart_speaker=None):
        self.nama = nama
        self.cpu = cpu if cpu else CPU()
        self.ram_list = ram_list if ram_list else []
        self.motherboard = motherboard if motherboard else Motherboard()
        self.ssd_list = ssd_list if ssd_list else []
        self.hdd_list = hdd_list if hdd_list else []
        self.monitor = monitor if monitor else Monitor()
        self.smart_speaker = smart_speaker if smart_speaker else SmartSpeaker()
    
    # Setter
    def set_nama(self, nama):
        self.nama = nama
    
    def set_cpu(self, cpu):
        self.cpu = cpu
    
    def set_ram_list(self, ram_list):
        self.ram_list = ram_list
    
    def set_motherboard(self, motherboard):
        self.motherboard = motherboard
    
    def set_ssd_list(self, ssd_list):
        self.ssd_list = ssd_list
    
    def set_hdd_list(self, hdd_list):
        self.hdd_list = hdd_list
    
    def set_monitor(self, monitor):
        self.monitor = monitor
    
    def set_smart_speaker(self, smart_speaker):
        self.smart_speaker = smart_speaker
    
    # Getter
    def get_nama(self):
        return self.nama
    
    def get_cpu(self):
        return self.cpu
    
    def get_ram_list(self):
        return self.ram_list
    
    def get_motherboard(self):
        return self.motherboard
    
    def get_ssd_list(self):
        return self.ssd_list
    
    def get_hdd_list(self):
        return self.hdd_list
    
    def get_monitor(self):
        return self.monitor
    
    def get_smart_speaker(self):
        return self.smart_speaker
    
    # Method untuk manipulasi komponen
    def tambah_ram(self, ram):
        self.ram_list.append(ram)
        print(f"RAM {ram.get_nama()} berhasil ditambahkan ke {self.nama}")
    
    def tambah_ssd(self, ssd):
        self.ssd_list.append(ssd)
        print(f"SSD {ssd.get_nama()} berhasil ditambahkan ke {self.nama}")
    
    def tambah_hdd(self, hdd):
        self.hdd_list.append(hdd)
        print(f"HDD {hdd.get_nama()} berhasil ditambahkan ke {self.nama}")
    
    # Tampilkan Info
    def tampilkan_spesifikasi(self):
        print(f"============ SPESIFIKASI KOMPUTER: {self.nama} ============\n")
        
        print(f"CPU: {self.cpu.tampilkan_info()}\n")
        
        print("RAM:")
        for i, ram in enumerate(self.ram_list, 1):
            print(f"  [{i}] {ram.tampilkan_info()}")
        print()
        
        print(f"MOTHERBOARD: {self.motherboard.tampilkan_info()}\n")
        
        print("STORAGE:")
        print("  SSD:")
        for i, ssd in enumerate(self.ssd_list, 1):
            print(f"    [{i}] {ssd.tampilkan_info()}")
        
        print("  HDD:")
        for i, hdd in enumerate(self.hdd_list, 1):
            print(f"    [{i}] {hdd.tampilkan_info()}")
        print()
        
        print(f"MONITOR: {self.monitor.tampilkan_info()}\n")
        
        print(f"SMART SPEAKER: {self.smart_speaker.tampilkan_info()}\n")
        
        print("=======================================================")